zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/processinstance/migration/MigrateExclusiveGatewayTest.java

```diff
@@ -0,0 +1,95 @@
+
+import static io.camunda.zeebe.engine.processing.processinstance.migration.MigrationTestUtil.extractProcessDefinitionKeyByProcessId;
+
+import io.camunda.zeebe.engine.util.EngineRule;
+import io.camunda.zeebe.model.bpmn.Bpmn;
+import io.camunda.zeebe.protocol.record.Assertions;
+import io.camunda.zeebe.protocol.record.intent.IncidentIntent;
+import io.camunda.zeebe.protocol.record.intent.ProcessInstanceIntent;
+import io.camunda.zeebe.protocol.record.value.BpmnElementType;
+import io.camunda.zeebe.test.util.BrokerClassRuleHelper;
+import io.camunda.zeebe.test.util.record.RecordingExporter;
+import io.camunda.zeebe.test.util.record.RecordingExporterTestWatcher;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TestWatcher;
+
+public class MigrateExclusiveGatewayTest {
+
+  @ClassRule public static final EngineRule ENGINE = EngineRule.singlePartition();
+  private static final String INVALID_EXPRESSION = "missing_var";
+
+  @Rule public final TestWatcher watcher = new RecordingExporterTestWatcher();
+  @Rule public final BrokerClassRuleHelper helper = new BrokerClassRuleHelper();
+
+  @Test
+  public void shouldResolveIncidentForInvalidCondition() {
+    final String sourceProcessId = helper.getBpmnProcessId();
+    final String targetProcessId = helper.getBpmnProcessId() + "_v2";
+
+    final var deployment =
+        ENGINE
+            .deployment()
+            .withXmlResource(
+                Bpmn.createExecutableProcess(sourceProcessId)
+                    .startEvent()
+                    .exclusiveGateway("xor")
+                    .conditionExpression(INVALID_EXPRESSION)
+                    .endEvent("A")
+                    .moveToLastGateway()
+                    .conditionExpression(INVALID_EXPRESSION)
+                    .endEvent("B")
+                    .done())
+            .withXmlResource(
+                Bpmn.createExecutableProcess(targetProcessId)
+                    .startEvent()
+                    .exclusiveGateway("xor2")
+                    .conditionExpression("true")
+                    .endEvent("A")
+                    .moveToLastGateway()
+                    .conditionExpression("false")
+                    .endEvent("B")
+                    .done())
+            .deploy();
+    final long targetProcessDefinitionKey =
+        extractProcessDefinitionKeyByProcessId(deployment, targetProcessId);
+
+    final long processInstanceKey =
+        ENGINE.processInstance().ofBpmnProcessId(sourceProcessId).create();
+
+    RecordingExporter.incidentRecords(IncidentIntent.CREATED)
+        .withProcessInstanceKey(processInstanceKey)
+        .withElementId("xor")
+        .await();
+
+    // when
+    ENGINE
+        .processInstance()
+        .withInstanceKey(processInstanceKey)
+        .migration()
+        .withTargetProcessDefinitionKey(targetProcessDefinitionKey)
+        .addMappingInstruction("xor", "xor2")
+        .migrate();
+
+    // then
```

Comment:
❌ We should await migration completion to avoid flakiness.

Reply:
💭 The `.migrate()` actually awaits the successful migration already, and the Incident is migrated as part of it. However, there's something to say for being explicit in tests, so we can add it. It's not necessary to avoid flakiness though.

Reply:
Ahh, we wait for successful expectation completion, right. Okay. 👍